src/api/controllers/commentController.js

```diff
@@ -0,0 +1,185 @@
+import _ from 'lodash';
+import Sequelize from 'sequelize';
+import status from '../../helpers/constants/status.codes';
+import models from '../models';
+import sendError from '../../helpers/error.sender';
+import errorMessages from '../../helpers/constants/error.messages';
+import generatePaginationDetails from '../../helpers/generate.pagination.details';
+
+const { Comment, User } = models;
+
+/**
+ *
+ *
+ * @class CommentController
+ */
+class CommentController {
+  /**
+   * A controller to post a new comment on an article
+   *
+   * @author Prémices
+   * @static
+   * @param {object} req - this is the request object
+   * @param {object} res - this is the response object
+   * @returns {object} - this function returns an object
+   * @memberof RatingsController
+   */
+  static async create(req, res) {
+    // Initializing variables
+    const { slug } = req.params;
+    const { body, parentCommentId } = req.body;
+    let createdComment;
+    const {
+      user: { id }
+    } = req.user;
+
+    try {
+      if (parentCommentId) {
+        const parentComment = await Comment.findOne({
+          where: {
+            id: parentCommentId,
+            parentCommentId: {
+              [Sequelize.Op.not]: null
+            }
+          }
+        });
+
+        if (parentComment) sendError(status.NOT_ALLOWED, res, 'parentCommentId', errorMessages.parentCommentId);
+        else {
+          createdComment = await Comment.create({
+            userId: id,
+            articleSlug: slug,
+            body,
+            parentCommentId
```

Comment:
@nkpremices what's the point of the `parentCommentId`?

Reply:
In case a user is replying to a comment,  `parentCommentId` will be the Id of the comment the user is replying to.